### PR TITLE
Develop Arange[SiliconFlow]

### PR DESCRIPTION
**Pull request by MARD1NO**
image

由于native_function内对arange有多个函数签名，因此注册了 arange_start, arange 两个：
image

### CODE REVIEW

File: benchmark/test_special_perf.py [hunk 1]
```diff
@@ -121,3 +121,32 @@ def padding_kwargs(dtype, batch, size):
...
+def test_perf_arange():
+    def arange_kwargs(dtype, batch, size):
+        # return {
+        #     "start": 0,
+        #     "step": 1,
+        #     "end": batch * size,
+        #     "device": "cuda",
+        #     "dtype": dtype,
+        # }
+
```
Comment (Bowen12992) [inline benchmark/test_special_perf.py]: delete this

**Review by Bowen12992 — DISMISSED** [review]
LGTM

File: src/flag_gems/ops/arange.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    step_offset = pid * BLOCK_SIZE * step
+
+    cols = tl.arange(0, BLOCK_SIZE)
+    arange_val = cols * step + step_offset + start
+    mask = cols + pid * BLOCK_SIZE
+    tl.store(y_ptr + cols, arange_val, mask=mask < size)
+
+
+def arange_start(
+    start, end, step=1, dtype=None, layout=None, device=None, pin_memory=None
```
Comment (StrongSpoon) [inline src/flag_gems/ops/arange.py]: the aten api is as `arange.start_step(Scalar start, Scalar end, Scalar step=1, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor`. I suggest keeping the star.

File: tests/test_special_ops.py [hunk 3]
```diff
@@ -374,3 +374,15 @@ def test_pad(shape, dtype, pad_mode, contiguous):
         res_out = torch.nn.functional.pad(ref_x, pad_params, pad_mode, pad_value)
 
     gems_assert_equal(ref_out, res_out)
+
+
+@pytest.mark.parametrize("start", [0, 1, 3])
+@pytest.mark.parametrize("step", [1, 2, 5])
+@pytest.mark.parametrize("end", [128, 256, 1024])
+@pytest.mark.parametrize("dtype", [torch.int64, torch.float32])
```
Comment (StrongSpoon) [inline tests/test_special_ops.py]: dtype could include all float types, and I suggest testing int types and None as well

File: src/flag_gems/ops/arange.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        device = torch.device(
+            "cuda"
+        )  # Note(Zhengzekang): Torch default value is CPU, but triton is target to GPU.
+
+    result = torch.empty((size,), device=device, dtype=dtype, pin_memory=pin_memory)
+    arange_func[grid,](result, start, end, step, size, BLOCK_SIZE)
+    return result
+
+
+def arange(end, step=1, dtype=None, layout=None, device=None, pin_memory=None):
```
Comment (StrongSpoon) [inline src/flag_gems/ops/arange.py]: ditto, api should be the same as aten

**Review by StrongSpoon — DISMISSED** [review]
lgtm

**Review by StrongSpoon — APPROVED** [review]
lgtm

**Pull request merged**